### PR TITLE
bingo-nosql: fix enumerateId (#220)

### PR DESCRIPTION
**Pull request by yuriiq**
Bugfix for #220

### CODE REVIEW

File: api/c/bingo-nosql/src/bingo_matcher.cpp [hunk 1]
```diff
@@ -1603,17 +1603,16 @@ bool MolGrossMatcher::_tryCurrent() /* const */
 EnumeratorMatcher::EnumeratorMatcher(BaseIndex& index) : BaseMatcher(index, (IndigoObject*&)_indigoObject)
 {
     _id_numbers = index.getIdMapping().size();
-    _current_id = 0;
+    _current_id = -1;
```
Comment (SunFellow) [inline api/c/bingo-nosql/src/bingo_matcher.cpp:1606]: No need to reinitialize a variable of the base class just initialized.

File: api/c/tests/unit/bingo-nosql/bingo_nosql_test.cpp [hunk 2]
```diff
@@ -32,6 +32,7 @@ TEST(BingoNosqlTest, test_enumerate_id)
     int e = bingoEnumerateId(db);
     while (bingoNext(e))
     {
+        ASSERT_EQ(count, bingoGetCurrentId(e));
```
Comment (SunFellow) [inline api/c/tests/unit/bingo-nosql/bingo_nosql_test.cpp:35]: Why we may rely upon the Id of the object in the enumerator must be equal to the sequential index?
This is nowhere specified.

File: api/java/bingo-nosql/src/test/java/com/epam/indigo/BingoTests.java [hunk 3]
```diff
@@ -23,4 +23,28 @@ void checkBingoVersion(@TempDir Path tempDir) {
...
+        Indigo indigo = new Indigo();
+        Bingo bingo = Bingo.createDatabaseFile(indigo, tempDir.toString(), "molecule");
+        bingo.insert(indigo.loadMolecule("C"));
+        bingo.insert(indigo.loadMolecule("CC"));
+        BingoObject enumerator = bingo.enumerateId();
+        int i = 0;
+        while (enumerator.next()) {
+            assertEquals(
+                i,
+                enumerator.getCurrentId(),
```
Comment (SunFellow) [inline api/java/bingo-nosql/src/test/java/com/epam/indigo/BingoTests.java:39]: The same here:
Why we may rely upon that `Id` of the object in the enumerator is equal to the sequential number?
This is nowhere specified.

File: api/java/bingo-nosql/src/test/java/com/epam/indigo/BingoTests.java [hunk 4]
```diff
@@ -23,4 +23,28 @@ void checkBingoVersion(@TempDir Path tempDir) {
...
+                "Checking currentId"
+            );
```
Comment (SunFellow) [inline api/java/bingo-nosql/src/test/java/com/epam/indigo/BingoTests.java:41]: The same here:
Why we may rely upon that `Id` of the object in the enumerator is equal to the sequential number?
This is nowhere specified.

File: api/c/bingo-nosql/src/bingo_matcher.cpp [hunk 5]
```diff
@@ -144,7 +144,7 @@ IndexCurrentReaction::~IndexCurrentReaction()
 BaseMatcher::BaseMatcher(BaseIndex& index, IndigoObject*& current_obj) : _index(index), _current_obj(current_obj)
 {
     _current_obj_used = false;
-    _current_id = 0;
+    _current_id = -1;
     _part_id = -1;
     _part_count = -1;
```
Comment (mkviatkovskii) [inline api/c/bingo-nosql/src/bingo_matcher.cpp:149]: Why do we init member fields here, not in member initializers list?

**Pull request merged**